client/src/components/mobile-menu/index.tsx

```diff
@@ -0,0 +1,57 @@
+'use client';
+
+import Link from 'next/link';
+
+import { X, ExternalLink } from 'lucide-react';
+
+import { cn } from '@/lib/classnames';
+
+import { useMobileMenu } from '@/store';
+
+import { modules, moduleColors } from '@/constants/modules';
+
+const MobileMenu = () => {
+  const [isOpen, setIsOpen] = useMobileMenu();
+  const toggleMenu = () => {
+    setIsOpen(!isOpen);
+  };
+  if (!isOpen) return null;
+  return (
+    <div className="absolute left-0 top-0 z-50 h-screen w-screen transform bg-gray-700 text-white animate-in slide-in-from-right-full">
```

Comment:
The focus management of this component is incorrect:
- when the user clicks the hamburger icon, the focus is not moved to this component
- the focus is not trapped within the component

I think that you need to wrap this component in a shadcn/ui dialog and all these issues will probably be fixed.